### DISCUSSION TITLE
"Year Statistic" naming

### DISCUSSION THREAD

**Discussion by wolfgang-ch** [mytourbook/mytourbook#1128] [Q&A]
I would like to rename the "Year Statistic" view into something, which is more related to it's content. Since it was introduced 2008, I was never really happy with it.

![image](https://github.com/mytourbook/mytourbook/assets/1283445/e2f560fd-2b94-46a0-a681-d7853a6a3eb1)

My suggesitons are "Reference Progress", "Reference Statistic" or do you have a better wording?

![image](https://github.com/mytourbook/mytourbook/assets/1283445/3653f9db-f3dd-4da8-81e6-70970ad69014)

**Comment by FJBDev** [comment]
Here are my suggestions:

- Timeline
- Reference Timeline
- Tour Reference Timeline

Whichever new name you end up choosing, can you make sure to rename the filename of the view to be consistent with the new name ?

In several places, i find it confusing that the filename is very different than the view names, examples:

[TourCatalogView.java](https://github.com/mytourbook/mytourbook/blob/37df84518934679b6f73cb789bf959a5c3c6d483/bundles/net.tourbook/plugin.xml#L562) for `Reference Tours View`

[RawDataView.java](https://github.com/mytourbook/mytourbook/blob/37df84518934679b6f73cb789bf959a5c3c6d483/bundles/net.tourbook/plugin.xml#L540) for `Data Import View`

etc.....

**Reply by wolfgang-ch** [reply]
> In several places, i find it confusing that the filename is very different than the view names, examples:
> 
> [TourCatalogView.java](https://github.com/mytourbook/mytourbook/blob/37df84518934679b6f73cb789bf959a5c3c6d483/bundles/net.tourbook/plugin.xml#L562) for `Reference Tours View`
> 
> [RawDataView.java](https://github.com/mytourbook/mytourbook/blob/37df84518934679b6f73cb789bf959a5c3c6d483/bundles/net.tourbook/plugin.xml#L540) for `Data Import View`
> 
> etc.....

Changing ID's is highly critical and causes the UI to not display the "old" view because this is done by it's id and this is stored in `workbench.xmi`. So changing any ID must be tested in the UI if it is working as before and this is just time consuming why I don't change ID's when not needed.

**Reply by wolfgang-ch** [reply]
> Here are my suggestions:
> 
>     * Timeline
> 
>     * Reference Timeline
> 
>     * Tour Reference Timeline
> 

Reference Timeline 👍

**Reply by FJBDev** [reply]
> Changing ID's is highly critical and causes the UI to not display the "old" view because this is done by it's id and this is stored in `workbench.xmi`. So changing any ID must be tested in the UI if it is working as before and this is just time consuming why I don't change ID's when not needed.

Is there a way to keep the same ID and just rename the `java` file name ?

**Comment by wolfgang-ch** [comment]
>Is there a way to keep the same ID and just rename the java file name ?

Renaming the java file name is not a problem by using the refactoring tools in Eclipse this also changes names in plugin.xml, ...

I already did some big renamings 2 weeks ago, TourCatalog as class is gone 😄 

https://github.com/mytourbook/mytourbook/commit/9c8f90d72803b0cc49c1d931ffd480584c334416

https://github.com/mytourbook/mytourbook/commit/dfb71499d76494bf6fe28b7ec4a1502668158f3c

https://github.com/mytourbook/mytourbook/commit/c5c9ba6a21c548bbdc17ef37c9f69dd7469e6555

**Comment by GurVJ** [comment]
In the french version, it is just named "Statistiques", why not only "Statistics" in English ?

**Reply by wolfgang-ch** [reply]
@GurVJ  This discussion is about the `Year Statistic`

![image](https://github.com/mytourbook/mytourbook/assets/1283445/121f3f22-0cfa-41f6-bf91-87497d825f34)

**Reply by GurVJ** [reply]
😚 Sorry I couldn't find this screen and it really looked like the Statistics one.

**Comment by wolfgang-ch** [comment]
@FJBDev 

After implementing [#1139](https://github.com/mytourbook/mytourbook/issues/1139), I think that we should rename this view again from `Reference Timeline` into "???", because it do not show only compared tours from a reference tour, it also shows geo compared tours from any tour.

My sugesstion for the new name is `Compared Timeline` or do you have a better suggestion

![image](https://github.com/mytourbook/mytourbook/assets/1283445/138f20ce-dd36-44a9-ad62-980b7200369d)

**Reply by FJBDev** [reply]
>My sugesstion for the new name is Compared Timeline or do you have a better suggestion

What about those suggestions:
- Geo Comparison Timeline
- Tour Comparison timeline

**Reply by FJBDev** [reply]
Another improvement I see would be to add a way to display pace instead of speed for people that do more running than cycling and hence use more min/km or min/mile as speed reference.

**Reply by wolfgang-ch** [reply]
Pace values can (should, when I didn't forget it somewhere) now be displayed everywhere in the tour compare views

**Comment by wolfgang-ch** [comment]
I changed the wording according to your suggestions. I've never done before to set the view name/icon programmatically but it works 😄 

View names and icons when **no compared** tour is selected

![image](https://github.com/mytourbook/mytourbook/assets/1283445/166f5565-805d-4659-a4d9-9c181eb991b9)

View names and icons when an **elevation compared** tour is selected

![image](https://github.com/mytourbook/mytourbook/assets/1283445/422989e1-bb31-47ff-8b9c-3ca70d1a86c0)


View names and icons when a **geo compared** tour is selected

![image](https://github.com/mytourbook/mytourbook/assets/1283445/49ba8af3-458d-44c6-81ef-5ecd1e57d629)
